internal/agent/config.go

```diff
@@ -85,7 +85,7 @@ type Config struct {
 	// DefaultLabels are automatically applied to this device when the agent is enrolled in a service
 	DefaultLabels map[string]string `json:"default-labels,omitempty"`
 
-	reader *fileio.Reader
+	reader fileio.Reader
```

Comment:
it is reader cause you don't want to allow it to write? Though at the end you create it with NewReadWriter?

Reply:
Same question actually, why it should be part of config and not part of the object ?

Reply:
correct it's an interface so as long as the interface fulfills Reader it is technically fine. And yeah the idea is to limit access to Writer where possible. But I did revert the change.

Reply:
to do what you want we could add a readWriter I limited it more because Write wasn't needed.

Reply:
Yeah but creating object in config is sort of abuse of it , it should be part of object that will actually use it, in anycase it is only IMO

Reply:
I am not a fan of how this is done so for sure open to ideas